tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/ApplyCondition.cs

```diff
@@ -0,0 +1,142 @@
+        /// </summary>
+        /// <param name="jsonElement">The contents of "condition" key with the body of a request to /Admin/AddSanitizer or /Admin/AddTransform.</param>
+        public ApplyCondition(JsonElement jsonElement)
+        {
+            var conditionDefined = false;
+            try
+            {
+                var uriProp = GetProp("UriRegex", jsonElement);
+
+                if(uriProp.Value.ValueKind != JsonValueKind.Undefined)
+                {
+                    UriRegex = uriProp.Value.GetString();
+                    conditionDefined = true;
+                }
+
+                // ... body/header condition support goes here.
+            }
+            catch(Exception e)
+            {
+                throw new HttpException(
+                    HttpStatusCode.BadRequest,
+                    $"An unexpected error occured during parse of condition body. Condition Definition: {jsonElement.GetRawText()}. Exception detail: {e.Message}."
+                );
+            }
+            
+            if (!conditionDefined)
+            {
+                throw new HttpException(
+                    HttpStatusCode.BadRequest, 
+                    $"This request defined a condition. The definition of said condition is invalid. At least one trigger regex must be present. Condition Definition: {jsonElement.GetRawText()}."
+                );
+            }
+        }
+
+        // sanitizers run against RecordEntries
+        public virtual bool Applicable(RecordEntry entry)
+        {
+            var result = Regex.Match(entry.RequestUri);
+            if (result.Success)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        // transforms are used directly on the request
+        public virtual bool Applicable(HttpRequest request)
+        {
+            var requestString = request.HttpContext.Features.Get<IHttpRequestFeature>().RawTarget;
+            var requestUri = request.Scheme + "://" + request.Host + requestString;
+
+            var result = Regex.Match(requestUri);
+            if (result.Success)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append("<pre>{");
+
+            sb.Append($"\n   \"UriRegex\": \"{_uriRegex}\",");
+            // ... body/header condition support goes here.
+
+            sb.Append("\n}</pre>");
+            return sb.ToString();
+        }
+    }
+
+    /// <summary>
+    /// This is the default condition in RecordedTestSanitizer. Utilizing this basic class, we can keep code calls simple,
+    /// without adding perf cost of a pointless forced evaluation.
+    /// </summary>
+    public class PassThruConditional : ApplyCondition
+    {
+        public PassThruConditional() : base(".+")
```

Comment:
I don't think we actually use the passed regex since we override the methods, right?